core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/babe/impl/babe_impl.hpp"
+
+#include <cmath>
+
+#include <sr25519/sr25519.h>
+#include <boost/assert.hpp>
+#include "consensus/babe/types/babe_block_header.hpp"
+#include "consensus/babe/types/seal.hpp"
+#include "libp2p/peer/peer_id.hpp"
+#include "network/types/block_announce.hpp"
+#include "scale/scale.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::consensus, BabeImpl::Error, e) {
+  using E = kagome::consensus::BabeImpl::Error;
+  switch (e) {
+    case E::TIMER_ERROR:
+      return "error happened while using the timer";
```

Comment:
Is it possible to put a more accurate error description? I think it's not quite clear what error may occur with a timer.